parsl/providers/provider_base.py

```diff
@@ -48,7 +87,7 @@ def submit(self, command: str, tasks_per_node: int, job_name: str = "parsl.auto"
         pass
 
     @abstractmethod
-    def status(self, job_ids: List[Any]) -> List[str]:
+    def status(self, job_ids: List[Any]) -> List[JobStatus]:
```

Comment:
Who's "we"?

ParslExector.status() does that.

It's fine here as a List because we can guarantee that it matches job_ids. Alternatively, we could have JobStatus have a job_id property.

The reason the executor is different is because, given the following hypothetical:

ids:List[str] = executor.get_job_ids()
states: List[JobStatus] = executor.status(),

it cannot be guaranteed that states matches ids since the job list internal to the executor may change. There is the additional issue that the job list may be constructed by the executor from a dict, which may or may not have predictable iteration order. Very recent versions of Python guarantee insertion order and slightly older versions of Python use insertion order, but do not officially guarantee it.